activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -568,9 +568,16 @@ def translate_exception_class(e, sql)
         end
 
         def log(sql, name = "SQL", binds = [], type_casted_binds = [], statement_name = nil) # :doc:
+          spec_name = if pool
+            pool.spec.name
+          else
+            ConnectionSpecification::Resolver.new(Base.configurations).spec(@config).name
```

Comment:
It is a shame we have to do this just because one place is trying to log before the connection pool is assigned to the connection.

Maybe we could refactor the code to do the query outside of the `initialize` method?

Let's wait to see if @matthewd or @tenderlove have some suggestion.